patterns/README.md

```diff
@@ -1030,3 +1030,371 @@ A field MAY have a `missingValues` property that MUST be an `array` where each e
+   pattern above, we propose that field-specific missing values be *added* to
+   the values appearing in the `missingValues` property at the resource level,
+   rather than replacing them. This is so that software can distinguish
+   between so-called *system missing values* (e.g., "Not applicable") and
+   other values that you may wish to include in certain tabulations/analyses
+   but exclude from others (e.g., "Don't know" or "Refused").
+2. Add an optional field-specific `ordered` property, which can be used when
+   contructing a categorical (or factor) to indicate that the variable is
+   ordinal.
+3. Add an optional field-specific `encoding` property for use when data are
+   stored using integer or other codes rather than using the category labels.
+   This contains an object mapping the codes appearing in the data (keys) to
+   what they mean (values), and can be used by software to construct
+   corresponding value labels or categoricals (when supported) or to translate
+   the values when reading the data.
+
+As none of the three proposed properties is part of the current
+[table schema](https://specs.frictionlessdata.io//table-schema/), the proposed
+extensions are fully backward compatible.
+
+Here is an example using extensions (1) and (2):
+
+```
+{
+  "fields": [
+    {
+      "name": "physical_health",
+      "type": "string",
+      "constraints": {
+        "enum": [
+          "Poor",
+          "Fair",
+          "Good",
+          "Very good",
+          "Excellent",
+        ]
+      }
+      "ordered": true
+      "missingValues": ["Don't know","Refused"]
+    }
+  ],
+  "missingValues": ["Not applicable","No answer"]
+}
+```
+
+This is our preferred strategy, as it provides all of the information
+necessary to support fully the categorical functionality of the software
+packages listed above, while still yielding a useable result for software
+without such capability. As described below, value labels or categoricals can
+be created automatically based on the ordering of the values in the `enum`
+array, and the field level `missingValues` can be incorporated into the value
+labels or categoricals if desired. In those cases where it is desired to have
+more control over how the value labels are constructed, this information can
+be stored in a separate encodings file in JSON format or as part of a custom
+extension to the table schema. Since such instructions do not describe the
+data themselves (but only how a specific software package should handle them),
+and since they are often software- and/or user-specific, we argue that they
+should not be included in the official table schema.
+
+Alternatively, those who wish to store their data in encoded form (e.g., this
+is the default for data exports from [REDCap](https://projectredcap.org), a
+commonly-used platform for collecting data for clinical studies) may use
+extension (3) to do so:
```

Comment:
The suggested `encoding` property is also useful for controlled vocabularies, to map between values and labels, so I think the suggested order `value`: `label` is great. I could envision this being extended to support multiple languages.